### PR TITLE
fix: add CLI library back into node

### PR DESCRIPTION
**Pull request by bhgomes**
Signed-off-by: Brandon H. Gomes <bhgomes@pm.me>

## Description

Adds back the CLI node library removed by this commit: https://github.com/Manta-Network/Manta/commit/59d1bc483574e67cedeeb28c7fb9c4805ea50669 needed for https://github.com/Manta-Network/cli/pull/1. Also, this PR resolves some formatting and documentation issues now covered by the CI.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests.
- [x] Updated relevant documentation in the code.
- [x] Added **one** line describing your change in `<branch>/CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [x] If runtime changes, need to update the version numbers properly:
   * `authoring_version`: The version of the authorship interface. An authoring node will not attempt to author blocks unless this is equal to its native runtime.
   * `spec_version`: The version of the runtime specification. A full node will not attempt to use its native runtime in substitute for the on-chain Wasm runtime unless all of spec_name, spec_version, and authoring_version are the same between Wasm and native.
   * `impl_version`: The version of the implementation of the specification. Nodes are free to ignore this; it serves only as an indication that the code is different; as long as the other two versions are the same then while the actual code may be different, it is nonetheless required to do the same thing. Non-consensus-breaking optimizations are about the only changes that could be made which would result in only the impl_version changing.
   * `transaction_version`: The version of the extrinsics interface. This number must be updated in the following circumstances: extrinsic parameters (number, order, or types) have been changed; extrinsics or pallets have been removed; or the pallet order in the construct_runtime! macro or extrinsic order in a pallet has been changed. You can run the `metadata_diff.yml` workflow for help. If this number is updated, then the `spec_version` must also be updated 
- [x] Verify benchmarks & weights have been updated for any modified runtime logics
- [x] If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- [x] If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- [x] If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.

### CODE REVIEW

**Comment by bhgomes** [comment]
Seems the `CHANGELOG.md` was not updated to include a section to add unreleased updates, so I'm going to wait for that.

**Review by ghzlatarev — DISMISSED** [review]

**Review by Garandor — COMMENTED** [review]
Pls add changelog and resolve conflicts

**Review by Garandor — COMMENTED** [review]
i don't see any functional changes in this PR, it seems to be type aliasing and refactoring only?

**Comment by bhgomes** [comment]
> i don't see any functional changes in this PR, it seems to be type aliasing and refactoring only?

The change is to expose the `node` code as a library which is necessary for the CLI project. There are no functional changes.

**Review by Garandor — APPROVED** [review]
LGTM!

**Review by Garandor — CHANGES_REQUESTED** [review]
taking that back, CI is failing with missing module documentation

**Comment by bhgomes** [comment]
> taking that back, CI is failing with missing module documentation

Yeah, fixing it now. The CI changed since last time and I wasn't ready for it yet.

**Review by Garandor — DISMISSED** [review]
LGTM!

File: node/src/chain_specs/calamari.rs [hunk 1]
```diff
@@ -14,21 +14,27 @@
...
-const KUSAMA_RELAYCHAIN_DEV_NET: &str = "kusama-dev";
+/// Kusama Relaychain Development Network Identifier
+pub const KUSAMA_RELAYCHAIN_DEV_NET: &str = "kusama-dev";
 
 /// The default XCM version to set in genesis config.
-const SAFE_XCM_VERSION: u32 = 2;
+pub const SAFE_XCM_VERSION: u32 = 2;
+
+/// Calamari Chain Spec
+pub type CalamariChainSpec = sc_service::GenericChainSpec<GenesisConfig, Extensions>;
```
Comment (Dengjianping) [inline node/src/chain_specs/calamari.rs:37]: ~~I suppose it can be reused, can it be move to mod.rs?~~

**Review by Dengjianping — DISMISSED** [review]
LGTM! But have to resolve the conflict. @bhgomes

**Review by Garandor — APPROVED** [review]

**Review by ghzlatarev — APPROVED** [review]

**Pull request merged**